### ISSUE TITLE
Slow training speed of Levenshtein Transformer in pre-training with translation_lev

### ISSUE THREAD

**Issue by h-sugi** [pytorch/fairseq#3127]
Hi! 
When I increased the number of nodes for the pre-training of Levenshtein transformer (from 100x V100 to 400x V100), wpb increased as the larger nodes but wps was not increased, so the training speed is still slow. 

#### Ex. of 100x V100
`mask_ins=1.533, word_ins=7.425, word_del=0.486, ppl=696.28, wps=110918, ups=0.31, wpb=356820, bsz=29030.6`
#### Ex. of 400x V100
`mask_ins=1.642, word_ins=8.158, word_del=0.552, ppl=1307.41, wps=128540, ups=0.09, wpb=1.38257e+06, bsz=112702`

The nat_loss or adding noises are the bottleneck of the performance? How to improve the training performance? Current training speed is too slow and the loss/ppl performance is too poor comparing to the standard transformer with the translation task (#2208).

Any helps? 

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.7.1
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8.5 (Anaconda3)
 - CUDA/cuDNN version: 10.2/8.0
 - GPU models and configuration: 100x or 400 x V100 16GB (4x V100 / node)
